docs/data/data-grid/column-dimensions/ColumnAutosizingAsync.tsx

```diff
@@ -0,0 +1,163 @@
+        rating: randomRating(),
+      }));
+
+      resolve({ rows });
+    }, 1000);
+  });
+}
+
+export default function ColumnAutosizingAsync() {
+  const apiRef = useGridApiRef<GridApiPro>();
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [rows] = React.useState([]);
+
+  const [includeHeaders, setIncludeHeaders] = React.useState(
+    DEFAULT_GRID_AUTOSIZE_OPTIONS.includeHeaders,
+  );
+  const [includeOutliers, setExcludeOutliers] = React.useState(
+    DEFAULT_GRID_AUTOSIZE_OPTIONS.includeOutliers,
+  );
+  const [outliersFactor, setOutliersFactor] = React.useState(
+    String(DEFAULT_GRID_AUTOSIZE_OPTIONS.outliersFactor),
+  );
+  const [expand, setExpand] = React.useState(DEFAULT_GRID_AUTOSIZE_OPTIONS.expand);
+
+  const autosizeOptions = React.useMemo(
+    () => ({
+      includeHeaders,
+      includeOutliers,
+      outliersFactor: Number.isNaN(parseFloat(outliersFactor))
+        ? 1
+        : parseFloat(outliersFactor),
+      expand,
+    }),
+    [expand, includeHeaders, includeOutliers, outliersFactor],
+  );
+
+  const fetchData = React.useCallback(
+    (options: typeof autosizeOptions) => {
+      setIsLoading(true);
+      getFakeData(100)
+        .then((data) => {
+          return ReactDOM.flushSync(() => {
+            setIsLoading(false);
+            apiRef.current.updateRows(data.rows);
+          });
+        })
+        .then(() => apiRef.current.autosizeColumns(options));
+    },
+    [apiRef],
+  );
+
+  React.useEffect(() => {
+    fetchData(DEFAULT_GRID_AUTOSIZE_OPTIONS);
+  }, [fetchData]);
+
+  return (
+    <div style={{ width: '100%' }}>
+      <Stack
+        spacing={1}
+        direction="row"
+        alignItems="center"
+        sx={{ mb: 1 }}
+        useFlexGap
+        flexWrap="wrap"
+      >
+        <Button variant="outlined" onClick={() => fetchData(autosizeOptions)}>
+          Refetch data
+        </Button>
+        <FormControlLabel
+          sx={{ ml: 0 }}
+          control={
+            <Checkbox
+              checked={includeHeaders}
+              onChange={(ev) => setIncludeHeaders(ev.target.checked)}
+            />
+          }
+          label="Include headers"
+        />
+        <FormControlLabel
+          sx={{ ml: 0 }}
+          control={
+            <Checkbox
+              checked={includeOutliers}
+              onChange={(event) => setExcludeOutliers(event.target.checked)}
+            />
+          }
+          label="Include outliers"
+        />
+        <TextField
+          size="small"
+          label="Outliers factor"
+          value={outliersFactor}
+          onChange={(ev) => setOutliersFactor(ev.target.value)}
+          sx={{ width: '12ch' }}
+        />
+        <FormControlLabel
+          sx={{ ml: 0 }}
+          control={
+            <Checkbox
+              checked={expand}
+              onChange={(ev) => setExpand(ev.target.checked)}
+            />
+          }
+          label="Expand"
+        />
+      </Stack>
```

Comment:
I feel like autizising options controls are redundant here.
Setting options to `{ includeHeaders: true, includeOutliers: true }` + "Refetch data" button should be enough, what do you think?